File: repair/back/controllers/rest/datacategory.go
package rest


import (
	"repair/controllers"
	"repair/models"

    "strings"
)

type DatacategoryController struct {
	controllers.Controller
}



func (c *DatacategoryController) Insert(item *models.Datacategory) {
    
    
	conn := c.NewConnection()
    
	manager := models.NewDatacategoryManager(conn)
	err := manager.Insert(item)
    if err != nil {
        c.Set("code", "error")    
        c.Set("error", err)
        return
    }

    id := manager.GetIdentity()
    c.Result["id"] = id
    item.Id = id
}

func (c *DatacategoryController) Insertbatch(item *[]models.Datacategory) {  
    if item == nil || len(*item) == 0 {
        return
    }

    rows := len(*item)
    
    
    
	conn := c.NewConnection()
    
	manager := models.NewDatacategoryManager(conn)

    for i := 0; i < rows; i++ {
	    err := manager.Insert(&((*item)[i]))
        if err != nil {
            c.Set("code", "error")    
            c.Set("error", err)
            return
        }
    }
}

func (c *DatacategoryController) Update(item *models.Datacategory) {
    
    
	conn := c.NewConnection()

	manager := models.NewDatacategoryManager(conn)
    err := manager.Update(item)
    if err != nil {
        c.Set("code", "error")    
        c.Set("error", err)
        return
    }
}

func (c *DatacategoryController) Delete(item *models.Datacategory) {
    
    
    conn := c.NewConnection()

	manager := models.NewDatacategoryManager(conn)

    
	err := manager.Delete(item.Id)
    if err != nil {
        c.Set("code", "error")    
        c.Set("error", err)
    }
}

func (c *DatacategoryController) Deletebatch(item *[]models.Datacategory) {
    
    
    conn := c.NewConnection()

	manager := models.NewDatacategoryManager(conn)

    for _, v := range *item {
        
    
	    err := manager.Delete(v.Id)
        if err != nil {
            c.Set("code", "error")    
            c.Set("error", err)
            return
        }
    }
}

func (c *DatacategoryController) Count() {
    
    
	conn := c.NewConnection()

	manager := models.NewDatacategoryManager(conn)

    var args []interface{}
    
    _name := c.Get("name")
    if _name != "" {
        args = append(args, models.Where{Column:"name", Value:_name, Compare:"="})
        
    }
    _category := c.Geti("category")
    if _category != 0 {
        args = append(args, models.Where{Column:"category", Value:_category, Compare:"="})    
    }
    _type := c.Geti("type")
    if _type != 0 {
        args = append(args, models.Where{Column:"type", Value:_type, Compare:"="})    
    }
    _remark := c.Get("remark")
    if _remark != "" {
        args = append(args, models.Where{Column:"remark", Value:_remark, Compare:"="})
    }
    _order := c.Geti("order")
    if _order != 0 {
        args = append(args, models.Where{Column:"order", Value:_order, Compare:"="})    
    }
    _startdate := c.Get("startdate")
    _enddate := c.Get("enddate")

    if _startdate != "" && _enddate != "" {        
        var v [2]string
        v[0] = _startdate
        v[1] = _enddate  
        args = append(args, models.Where{Column:"date", Value:v, Compare:"between"})    
    } else if  _startdate != "" {          
        args = append(args, models.Where{Column:"date", Value:_startdate, Compare:">="})
    } else if  _enddate != "" {          
        args = append(args, models.Where{Column:"date", Value:_enddate, Compare:"<="})            
    }
    

    
    
    total := manager.Count(args)
	c.Set("total", total)
}


func (c *DatacategoryController) Read(id int64) {
    
    
	conn := c.NewConnection()

	manager := models.NewDatacategoryManager(conn)
	item := manager.Get(id)

    
    
    c.Set("item", item)
}

func (c *DatacategoryController) Index(page int, pagesize int) {
    
    
	conn := c.NewConnection()

	manager := models.NewDatacategoryManager(conn)

    var args []interface{}
    
    _name := c.Get("name")
    if _name != "" {
        args = append(args, models.Where{Column:"name", Value:_name, Compare:"="})
        
    }
    _category := c.Geti("category")
    if _category != 0 {
        args = append(args, models.Where{Column:"category", Value:_category, Compare:"="})    
    }
    _type := c.Geti("type")
    if _type != 0 {
        args = append(args, models.Where{Column:"type", Value:_type, Compare:"="})    
    }
    _remark := c.Get("remark")
    if _remark != "" {
        args = append(args, models.Where{Column:"remark", Value:_remark, Compare:"="})
    }
    _order := c.Geti("order")
    if _order != 0 {
        args = append(args, models.Where{Column:"order", Value:_order, Compare:"="})    
    }
    _startdate := c.Get("startdate")
    _enddate := c.Get("enddate")
    if _startdate != "" && _enddate != "" {        
        var v [2]string
        v[0] = _startdate
        v[1] = _enddate  
        args = append(args, models.Where{Column:"date", Value:v, Compare:"between"})    
    } else if  _startdate != "" {          
        args = append(args, models.Where{Column:"date", Value:_startdate, Compare:">="})
    } else if  _enddate != "" {          
        args = append(args, models.Where{Column:"date", Value:_enddate, Compare:"<="})            
    }
    

    
    
    if page != 0 && pagesize != 0 {
        args = append(args, models.Paging(page, pagesize))
    }
    
    orderby := c.Get("orderby")
    if orderby == "" {
        if page != 0 && pagesize != 0 {
            orderby = "id desc"
            args = append(args, models.Ordering(orderby))
        }
    } else {
        orderbys := strings.Split(orderby, ",")

        str := ""
        for i, v := range orderbys {
            if i == 0 {
                str += v
            } else {
                if strings.Contains(v, "_") {                   
                    str += ", " + strings.Trim(v, " ")
                } else {
                    str += ", dc_" + strings.Trim(v, " ")                
                }
            }
        }
        
        args = append(args, models.Ordering(str))
    }
    
	items := manager.Find(args)
	c.Set("items", items)

    if page == 1 {
       total := manager.Count(args)
	   c.Set("total", total)
    }
}

File: repair/periodic/lib/models/datacategory.dart
import 'package:common_control/common_control.dart';


  
class Datacategory { 
  int id;
  String name;
  int category;
  int type;
  String remark;
  int order;
  String date;
  bool checked;
  Map<String, dynamic> extra;  

  Datacategory({        
          this.id = 0,       
          this.name = '',       
          this.category = 0,       
          this.type = 0,       
          this.remark = '',       
          this.order = 0,       
          this.date = '',
          this.extra = const{},
          this.checked = false}) ;
  

  factory Datacategory.fromJson(Map<String, dynamic> json) {
    return Datacategory(
        id: json['id'] as int,
        name: json['name'] as String,
        category: json['category'] as int,
        type: json['type'] as int,
        remark: json['remark'] as String,
        order: json['order'] as int,
        date: json['date'] as String, 
        extra: json['extra'] == null ? <String, dynamic>{} : json['extra'] as Map<String, dynamic>        
    );
  }

  Map<String, dynamic> toJson() =>
      { 'id': id,'name': name,'category': category,'type': type,'remark': remark,'order': order,'date': date };
}

class DatacategoryManager {
  static const baseUrl = '/api/datacategory';  

  static Future<List<Datacategory>> find(
      {int page = 0, int pagesize = 20, String? params}) async {
    var result =
        await Http.get(baseUrl, {'page': page, 'pagesize': pagesize}, params);
    if (result == null || result['items'] == null) {
      return List<Datacategory>.empty(growable: true);
    }

    return result['items']
        .map<Datacategory>((json) => Datacategory.fromJson(json))
        .toList();
  }

  static Future<Datacategory> get(int id) async {
    var result = await Http.get('$baseUrl/$id');
    if (result == null || result['item'] == null) {
      return Datacategory();
    }

    return Datacategory.fromJson(result['item']);
  }

  static Future<int> insert(Datacategory item) async {
    var result = await Http.insert(baseUrl, item.toJson());
    return result;
  }

  static update(Datacategory item) async {
    await Http.put(baseUrl, item.toJson());
  }

  static delete(Datacategory item) async {
    await Http.delete(baseUrl, item.toJson());
  }
}
